flang/include/flang/Semantics/scope.h

```diff
@@ -61,7 +61,7 @@ class Scope {
 public:
   ENUM_CLASS(Kind, Global, IntrinsicModules, Module, MainProgram, Subprogram,
       BlockData, DerivedType, BlockConstruct, Forall, OtherConstruct,
-      OpenACCConstruct, ImpliedDos, OtherClause)
+      OpenACCConstruct, ImpliedDos, OtherClause, CUFKernelDoConstruct)
```

Comment:
You could perhaps avoid some complexity by considering `!$CUF KERNEL DO` to be an OpenAcc construct.

Reply:
Yeah that works. I removed the special kind for cuf kernel.